src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryMock.java

```diff
@@ -728,4 +737,17 @@ public void setCustomTitle(@Nullable Component customTitle)
 		this.customTitle = customTitle;
 	}
 
+	@Override
+	public boolean equals(Object o)
+	{
+		if (!(o instanceof InventoryMock that)) return false;
+		return maxStackSize == that.maxStackSize && Objects.deepEquals(items, that.items) && Objects.equals(holder, that.holder) && type == that.type && Objects.equals(viewers, that.viewers) && Objects.equals(customTitle, that.customTitle);
+	}
+
+	@Override
+	public int hashCode()
+	{
+		return Objects.hash(Arrays.hashCode(items), holder, type, viewers, maxStackSize, customTitle);
+	}
+
```

Comment:
I'm not sure if i should include this. Personally i think this is a nice feature to have when validating inventories, but at the same time its a different behavior from Paper.

Reply:
This has been reverted, and a new method _isSimilar_ was added  at: https://github.com/MockBukkit/MockBukkit/blob/b1d4c3e6f0cacd3b726f396821175cb67336d355/src/main/java/org/mockbukkit/mockbukkit/inventory/InventoryMock.java#L740-L769